mpstat.c

```diff
@@ -2199,6 +2203,12 @@ void rw_mpstat_loop(int dis_hdr, int rows)
 				curr ^= 1;
 			}
 		}
+
+		/* -H option given and vCPU physical hotplug occured */
+		if (USE_OPTION_H(flags) && cpu_nr < get_cpu_nr(~0, TRUE)) {
```

Comment:
To avoid looking for number of available CPU several times (here and in `mpstat_loop()` function), you can:
1) Make function `rw_mpstat_loop()` return an int
2) look for the number of available CPU and save its result in a new variable (e.g. new_cpu_nr).
3) if cpu_nr < new_cpu_nr then return new_cpu_nr.
4) return 0 as a default value in all other cases (and in particular at the end of the function `rw_mpstat_loop()`).
Variable photplug is no longer necessary and can be removed (a non zero value retuned by `rw_mpstat_loop()` means a vCPU has been physically hotplugged).

Reply:
Taken care of all the above points:
1. rw_mpstat_loop() returns an int
2. new_cpu_nr saved in rw_mpstat_loop
3. if cpu_nr < new_cpu_nr : return new_cpu_nr
4. return 0 as a default value (at end and when SIGINT recevied in first interval) and photplug variable removed